service/javadsl/server/src/main/java/com/lightbend/lagom/javadsl/server/ServiceGuiceSupport.java

```diff
@@ -16,9 +16,21 @@
 
 import java.util.Arrays;
 
+/**
+ * Lagom service implementations must create one implementation of this interface and use it to bind Service
+ * implementations.
+ */
 public interface ServiceGuiceSupport extends ServiceClientGuiceSupport {
 
-    // redefines bindServiceInfo from super interface.
+    /**
+     * Creates a custom {@link ServiceInfo} for this Lagom service. This method overrides
+     * {@link ServiceClientGuiceSupport#bindServiceInfo(ServiceInfo)} with custom behavior for consume-only Lagom
+     * services.
+     * <p>
+     * The customization may be used by Lagom to add features only applicatble to Lagom services transparently.
```

Comment:
Got your point wrt MetricsService.
I actually didn't like that the sentence "The customization may be used by Lagom to add features only applicatble to Lagom services transparently." was only included in `bindServiceInfo`.

I used a different approach now (see upcoming commit):

 * `bindServiceInfo`s javadoc states it's overriding because it's a service not a client and may need specifics
 * class javadocs include more info wrt restrictions (invoked either method exactly once) and introduce how those methods introduce features transparently.

I got the idea from `HeaderFilter` where the class javadoc explains the purpose and use cases and the method javadocs focus on the method alone.